web/ui/dashboard/src/app/private/pages/project/apps/app-details/cli/cli.component.html

```diff
@@ -0,0 +1,74 @@
+<convoy-card className="mb-20px min-h-[50vh]">
+
+    <!-- tab  -->
+	<div class="flex items-center justify-center border-b border-grey-10">
+		<ul class="w-full flex items-center">
+			<li *ngFor="let tab of tabs" (click)="toggleActiveTab(tab)">
+				<convoy-button
+					type="clear"
+					color="grey"
+					[className]="
+						activeTab === tab
+							? 'rounded-[0] py-16px px-18px border-b-[3px] border-transparent [border-bottom-style:solid] transition-all ease-in-out border-b-primary-100'
+							: 'rounded-[0] py-16px px-18px border-b-[3px] border-transparent [border-bottom-style:solid] transition-all ease-in-out'
+					"
+				>
+					<span [class]="activeTab === tab ? 'text-primary-100 capitalize text-14 font-semibold' : 'capitalize text-14'">{{ tab }}</span>
+				</convoy-button>
+			</li>
+
+			<div class="text-grey-40 text-14 font-medium whitespace-nowrap ml-auto pr-14px">
+				<convoy-button size="sm" type="outline" (click)="generateNewKey()">Generate New Key</convoy-button>
+			</div>
+		</ul>
+	</div>
+
+	<!-- empty state for both devices and api keys  -->
+	<ng-container *ngIf="(activeTab === 'api keys' && apiKeys?.length === 0) || (activeTab === 'devices' && devices?.length === 0)">
+		<convoy-empty-state imgSrc="/assets/img/empty-state.svg" [heading]="activeTab === 'api keys' ? 'No api keys have been generated for this application' : 'No devices have been registered under this application'" className="h-[50vh]" id="empty-state"></convoy-empty-state>
```

Comment:
Let's use fewer words for this empty state like: 'No API keys yet' and 'No CLI device yet'